t/test_cleanupSpies

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+readonly TEST_DIR="$(dirname "$0")"
+
+# shellcheck source=/dev/null
+. "$TEST_DIR/../shpy"
+# shellcheck source=/dev/null
+. "$TEST_DIR/../shpy-shunit2"
+
+testRemovesDirectory() {
+  createSpy foo
+
+  # shellcheck disable=SC2154
+  assertNotNull "${_shpy_spies_dir}"
+  [ -d "${_shpy_spies_dir}" ] || fail "Temporary directory does not exist"
+
+  cleanupSpies
+
+  [ ! -d "${_shpy_spies_dir}" ] || fail "Temporary directory should have been removed"
+}
+
+testWorksUnderNoUnsetOption() {
```

Comment:
This can be removed, my upcoming PR runs all tests with nounset to ensure shpy runs under those stricter conditions